### PR TITLE
Ab#81965 change context datasource selection

### PR DESCRIPTION
**Pull request by MwanPygmay**
# Description

Change layout for context datasource selection.
Fix translation from 'reference data' to 'select a reference data'.

## Useful links

- [Please insert link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/81965)

## Type of change

- [x] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

Go in the context datasource tab and try to edit it.

## Screenshots

context_datasource

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by unai-reliefapp** [comment]
This is the ticket right @MwanPygmay  ? https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/81965
Can you change it in the PR description and rename the branch so the PR is linked to the correct ticket please?

Danke

File: apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts [hunk 1]
```diff
@@ -132,11 +120,12 @@ export class ContextDatasourceComponent
...
 
-        if (value) displayField?.enable();
-        else displayField?.disable();
         if (value) {
+          this.form.controls.refData.setValue(null);
+          this.refData = null;
+          displayField?.enable();
           this.getResource(value);
-        }
+        } else displayField?.disable();
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts]:
```suggestion
        } else {
        displayField?.disable();
        }
```
Ugly but helps debugging

File: apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts [hunk 2]
```diff
@@ -152,11 +141,12 @@ export class ContextDatasourceComponent
...
 
-        if (value) displayField?.enable();
-        else displayField?.disable();
         if (value) {
+          this.form.controls.resource.setValue(null);
+          this.resource = null;
           this.getReferenceData(value);
-        }
+          displayField?.enable();
+        } else displayField?.disable();
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts]:
```suggestion
        } else {
        displayField?.disable();
        }
```
Ugly but helps debugging

File: apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts [hunk 3]
```diff
@@ -207,46 +197,52 @@ export class ContextDatasourceComponent
...
-      case 'refData':
-        // TODO: When frontend changes about referenceData fields are merged,
-        // swap to the commented line to remove any casting
-        // this.availableFields = this.refData?.fields?.map((x) => x.name) ?? [];
-        this.availableFields =
-          this.refData?.fields?.map((x: any) => x.name) ?? [];
-        break;
-      default:
-        this.availableFields = [];
+    console.log();
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/dashboard/components/context-datasource/context-datasource.component.ts]: image

```suggestion
```

**Review by unai-reliefapp — CHANGES_REQUESTED** [review]

**Comment by unai-reliefapp** [comment]
@AntoineRelief would it be worth it to refactor this piece of code as it's use in lot of places same as we did with the `shared-filter-list` component? https://github.com/ReliefApplications/ems-frontend/blob/beta/libs/shared/src/lib/components/list-filter/list-filter.component.ts#L47-L52

And set those fields by default and add extend the type when creating those forms?
Or maybe in other ticket

KR, Unai

**Review by unai-reliefapp — APPROVED** [review]

**Comment by AntoineRelief** [comment]
> @AntoineRelief would it be worth it to refactor this piece of code as it's use in lot of places same as we did with the `shared-filter-list` component? https://github.com/ReliefApplications/ems-frontend/blob/beta/libs/shared/src/lib/components/list-filter/list-filter.component.ts#L47-L52
> 
> And set those fields by default and add extend the type when creating those forms? Or maybe in other ticket
> 
> KR, Unai

@unai-reliefapp 
in another ticket, I only want to make it work in this one

**Comment by unai-reliefapp** [comment]
Remember to change branch name!!!! @MwanPygmay 
It's 81965 not 81695
Approved 👍

**Comment by AntoineRelief** [comment]
@unai-reliefapp @MwanPygmay 
all good
only a small lint issue 👍

**Comment by MwanPygmay** [comment]
@unai-reliefapp  oopsi yeah did only change ticket in description sorry!

**Pull request merged**